packages/algob/test/internal/deployer.ts

```diff
@@ -112,18 +114,76 @@ describe("DeployerDeployMode", () => {
     const deployer = new DeployerDeployMode(deployerCfg);
 
     const asaInfo = await deployer.deployASA("MY_ASA", { creator: deployer.accounts[0] });
-    assert.deepEqual(asaInfo, { creator: "addr-1-get-address-dry-run", txId: "tx-id-dry-run", confirmedRound: -1, assetIndex: -1 });
+    assert.deepEqual(asaInfo,
+      { creator: "addr-1-get-address-dry-run", txId: "tx-id-dry-run", confirmedRound: -1, assetIndex: -1, assetDef: mkASA() });
 
     deployerCfg.cpData.precedingCP.network1.timestamp = 515236;
-    // console.log(deployerCfg.cpData.precedingCP);
     assert.deepEqual(deployerCfg.cpData.precedingCP, {
       network1: {
+        asa: new Map([["MY_ASA", {
+          creator: "addr-1-get-address-dry-run",
+          txId: "tx-id-dry-run",
+          confirmedRound: -1,
+          assetIndex: -1,
+          assetDef: mkASA()
+        }]]),
+        ssc: new Map(),
+        dLsig: new Map(),
+        metadata: new Map<string, string>(),
+        timestamp: 515236
+      }
+    });
+  });
+
+  it("Should save overriden asaDef to checkpoint after asset deployment if custom ASA params are passed", async () => {
+    const env = mkEnv("network1");
+    const deployerCfg = new DeployerConfig(env, new AlgoOperatorDryRunImpl());
+    const accounts = genAccounts(4);
+    const fixedAccount = generateAccount();
+    deployerCfg.asaDefs = {
+      MY_ASA: {
+        ...mkASA(),
+        manager: accounts[0].addr,
+        reserve: accounts[1].addr,
+        clawback: fixedAccount.addr,
+        freeze: fixedAccount.addr
+      }
+    };
+    const deployer = new DeployerDeployMode(deployerCfg);
+
+    // passing different manager & reserve address in customParams during ASA deploy
+    const asaInfo = await deployer.deployASA("MY_ASA", { creator: deployer.accounts[0] }, {
+      manager: accounts[2].addr,
+      reserve: accounts[3].addr
+    });
+
+    // manager, reserve should be overriden
+    // clawback, freeze should be original one
+    const expectedASADef = {
+      ...mkASA(),
+      manager: accounts[2].addr,
+      reserve: accounts[3].addr,
+      clawback: fixedAccount.addr,
+      freeze: fixedAccount.addr
+    };
+
+    assert.deepEqual(asaInfo, {
+      creator: "addr-1-get-address-dry-run",
+      txId: "tx-id-dry-run",
+      confirmedRound: -1,
+      assetIndex: -1,
+      assetDef: expectedASADef
+    });
 
```

Comment:
Let's add in this test the following check
* recreate `deployer`
* check that `deployer.asa.get("MY_ASA")` equals to `expectedASADef`.

Reply:
Added.